portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -95,6 +98,19 @@ private String getComment(HttpServletRequest request) {
     return request.getParameter("comment");
   }
 
+  //Get request number from query string
+  private int getRequestNum(HttpServletRequest request) {
+    String requestString = request.getParameter("request");
+    int requestNum;
+    try {
+      requestNum = Integer.parseInt(requestString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + requestString);
+      return -1;
```

Comment:
So in the failure case, your page will have a single comment. (Since it will add a comment, realize it's added more than necessary at the break, and then stop adding comments). Not sure if that was what was intended, but think about whether you want the default behavior to be no comments or a small default number of comments.

Reply:
Good point! Thanks for catching it.